File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1
        while left <= right:
            if nums[(left + right)//2] == target:
                return (left + right)//2
            elif nums[(left + right)//2] < target:
                left = (left + right)//2 +1
            else:
                right = (left + right)//2 -1
        return -1

File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int left = 0;
        int right = n-1, maxarea = 0;
        while (left<right){
            int l = min(heights[left],heights[right]);
            int b = right -left;
            maxarea =max(maxarea,l*b);
            if (heights[left] < heights[right]){
                left ++;
            }
            else right--;
        }
        return maxarea;
    }
};
